Source/driver/Castro.cpp

```diff
@@ -954,6 +954,41 @@ Castro::initData ()
 
        }
 
+       // it is not a requirement that the problem setup defines the
+       // temperature, so we do that here _and_ ensure that we are
+       // within any small limits
+       computeTemp(S_new, cur_time, S_new.nGrow());
+
+#ifdef _OPENMP
+#pragma omp parallel
+#endif
+       for (MFIter mfi(S_new, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+         {
+           const Box& bx = mfi.tilebox();
+
+           auto S_arr = S_new.array(mfi);
+
+           Real lsmall_temp = small_temp;
+           Real lsmall_dens = small_dens;
+
+           AMREX_PARALLEL_FOR_3D(bx, i, j, k,
+           {
+
+             // if the problem tried to initialize a thermodynamic
+             // state that is at or below small_temp, then we abort.
+             // This is dangerous and we should recommend a smaller
+             // small_temp
+             if (S_arr(i,j,k,UTEMP) < lsmall_temp * 1.001) {
+               amrex::Error("Error: initial data has T <~ small_temp");
```

Comment:
To work as desired in the CUDA build, we should do this as a max reduction on an integer flag that starts as 0 and is set to 1 if the floor is violated, then throw the error outside the parallel for.

Reply:
can you point me to an example?

Reply:
Yes, look at what we do with `burn_failed` in `react_state`.